src/main/java/picard/illumina/parser/IlluminaDataProviderFactory.java

```diff
@@ -144,19 +138,19 @@ public IlluminaDataProviderFactory(final File basecallDirectory, final int lane,
     /**
      * Create factory with the specified options, one that favors using QSeqs over all other files
      *
-     * @param basecallDirectory The baseCalls directory of a complete Illumina directory.  Files are found by searching relative to this folder (some of them higher up in the directory tree).
-     * @param barcodesDirectory The barcodesDirectory with barcode files extracted by 'ExtractIlluminaBarcodes' (optional, use basecallDirectory if not specified)
-     * @param lane              Which lane to iterate over.
-     * @param readStructure     The read structure to which output clusters will conform.  When not using QSeqs, EAMSS masking(see BclParser) is run on individual reads as found in the readStructure, if
-     *                          the readStructure specified does not match the readStructure implied by the sequencer's output than the quality scores output may differ than what would be found
-     *                          in a run's QSeq files
-     * @param dataTypesArg      Which data types to read
+     * @param basecallDirectory            The baseCalls directory of a complete Illumina directory.  Files are found by searching relative to this folder (some of them higher up in the directory tree).
+     * @param barcodesDirectory            The barcodesDirectory with barcode files extracted by 'ExtractIlluminaBarcodes' (optional, use basecallDirectory if not specified)
```

Comment:
barcodeDirectory is not optional (yet), but this is meant to document the fact that if you don't set one it will be set to `basecallsDirectory` by default.

I'll try to reword it.